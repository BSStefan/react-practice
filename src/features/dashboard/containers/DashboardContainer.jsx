// @flow
import React from 'react';
import styled from 'styled-components';
import { Jumbotron } from 'react-bootstrap';

const DashboardContainer = (props: {}) => {
  return (
    <AwesomeDashboard>
      <Jumbotron>
        <h1>Movie DB App</h1>
        <p>
          This is an amaizing movie database application.
        </p>
        <p>
          Also this app will teach you how to write unit tests.
        </p>
      </Jumbotron>
    </AwesomeDashboard>
  )
};

const AwesomeDashboard = styled.div`
  text-align: center;
  width: 100%;
  height: 100vh;
  padding: 10%;
`

export default DashboardContainer;


