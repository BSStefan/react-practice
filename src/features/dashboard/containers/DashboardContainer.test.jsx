// @flow
import * as React from 'react';
import { shallow } from 'enzyme';
import DashboardContainer from './DashboardContainer';
import { Jumbotron } from 'react-bootstrap';
import renderer from 'react-test-renderer';

it('render component successfully', () => {
  const wrapper = shallow(<DashboardContainer />);

  expect(wrapper.exists()).toBe(true);
})

it('should find h1 tag with appropriate text', () => {
  const wrapper = shallow(<DashboardContainer />);

  const h1Tags = wrapper.find('h1')

  expect(h1Tags.text()).toBe("Movie DB App");
})

it('should find 2 p tags', () => {
  const wrapper = shallow(<DashboardContainer />);

  const pTags = wrapper.find('p')

  expect(pTags.length).toBe(2);
})

it('should find first p tag with appropriate text', () => {
  const wrapper = shallow(<DashboardContainer />);

  const h1First = wrapper.find('p').first()

  expect(h1First.text()).toBe("This is an amaizing movie database application.");
})

it('should find second p tag with appropriate text', () => {
  const wrapper = shallow(<DashboardContainer />);

  const h1Second = wrapper.find('p').first()

  expect(h1Second.text()).toBe("This is an amaizing movie database application.");
})

it('should match snapshot', () => {
  const tree = renderer
    .create(<DashboardContainer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})