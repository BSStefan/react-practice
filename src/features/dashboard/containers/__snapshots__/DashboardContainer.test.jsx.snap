// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="sc-bdVaJa hLyHSZ"
>
  <div
    className="jumbotron"
  >
    <h1>
      Movie DB App
    </h1>
    <p>
      This is an amaizing movie database application.
    </p>
    <p>
      Also this app will teach you how to write unit tests.
    </p>
  </div>
</div>
`;
